main.tf

```diff
@@ -64,15 +70,18 @@ resource "aws_subnet" "private_restricted" {
   vpc_id            = aws_vpc.vpc.id
   availability_zone = element(var.azs, count.index)
   cidr_block        = lookup(var.private_restricted_subnets[count.index], "cidr")
-  tags              = merge(var.tags, map("Name", format("%s-%s-%s", var.name, lookup(var.private_restricted_subnets[count.index], "name") , substr(element(var.azs, count.index), -1, 1))))
+  tags = merge(var.tags, {
+    "Name"         = format("%s-%s-%s", var.name, lookup(var.private_restricted_subnets[count.index], "name"), substr(element(var.azs, count.index), -1, 1))
+    "network_type" = lookup(var.private_restricted_subnets[count.index], "name")
```

Comment:
Can we update the tag delimiter to a hyphen instead of an underscore?
The hyphen models after aws's own tagging standard mentioned in:
https://d1.awsstatic.com/whitepapers/aws-tagging-best-practices.pdf